### PR TITLE
Components: Refactor `ColorPalette` tests to `@testing-library/react`

### PR DESCRIPTION
**Pull request by tyxla**
## What?
We've recently started refactoring `enzyme` tests to `@testing-library/react`.

This PR refactors the `ColorPalette` tests from `enzyme` to `@testing-library/react`.

## Why?
`@testing-library/react` provides a better way to write tests for accessible components that is closer to the way the user experiences them.

## How?
We're straightforwardly replacing `enzyme` tests with `@testing-library/react` ones, using `jest-dom` matchers and mocks to avoid testing unrelated implementation details.

Ideally, we should not be using snapshot tests here, however, the purpose of this PR is not to improve or enrich the tests themselves, but rather to migrate them `@testing-library/react`. Our primary motivation is unblocking the upgrade to React 18. 

In any case, it would be a good future step to improve the tests and test actual experience and interaction in order to improve the overall value of the tests.

## Testing Instructions
Verify tests pass: `npm run test:unit packages/components/src/color-palette/test/index.js`

### CODE REVIEW

File: packages/components/src/color-palette/test/index.js [hunk 1]
```diff
@@ -17,102 +18,146 @@ describe( 'ColorPalette', () => {
...
+		);
+
+		await user.click(
+			screen.getAllByRole( 'button', {
+				name: /^Color:/,
+				pressed: false,
+			} )[ 0 ]
+		);
 
 		expect( onChange ).toHaveBeenCalledTimes( 1 );
```
Comment (ciampo) [inline packages/components/src/color-palette/test/index.js:92]: To make this test slightly better, we could check that `onChange` was called with the value of the color that was clicked?

File: packages/components/src/color-palette/test/index.js [hunk 2]
```diff
@@ -17,102 +18,146 @@ describe( 'ColorPalette', () => {
...
+			<ColorPalette
+				colors={ colors }
+				value={ currentColor }
+				onChange={ onChange }
+			/>
+		);
 
-			test( 'should call onToggle on click.', () => {
-				renderedToggleButton.find( 'button' ).simulate( 'click' );
+		await user.click(
```
Comment (ciampo) [inline packages/components/src/color-palette/test/index.js:141]: Should we add a spy for the `onToggle` callback prop and make sure that `.toHaveBeenCalledTimes( 1 )`, like in the original test?

File: packages/components/src/color-palette/test/index.js (hunk 1 above)
Comment (tyxla) [reply packages/components/src/color-palette/test/index.js:92]: Good call - done in 833aa6f8d5bbdd3aaeb496f1f3928d6f7974f6fb

File: packages/components/src/color-palette/test/index.js (hunk 2 above)
Comment (tyxla) [reply packages/components/src/color-palette/test/index.js:141]: Hm, how would you recommend this to happen without modifications to the original component?

Comment (ciampo) [reply packages/components/src/color-palette/test/index.js:141]: My bad, I didn't realize that the `Dropdown` component is an implementation detail, internal to `ColorPalette` and therefore it's not possible to pass a spy for `onToggle` 😅

**Review by ciampo — APPROVED** [review]
🚀

**Pull request merged**